cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/type/DatasetTypeManager.java

```diff
@@ -376,6 +378,27 @@ private void deployDefaultModules() {
     }
   }
 
+  private void deleteSystemModules() {
+    try {
+      mdsDatasets.execute(new TxCallable<MDSDatasets, Void>() {
+        @Override
+        public Void call(MDSDatasets context) throws Exception {
+          DatasetTypeMDS typeMDS = context.getTypeMDS();
+          Collection<DatasetModuleMeta> allDatasets = typeMDS.getModules(Constants.SYSTEM_NAMESPACE_ID);
+          for (DatasetModuleMeta ds : allDatasets) {
+            if (ds.getJarLocation() == null) {
```

Comment:
Possibly in the future, if we allow other developers to plug in their own "system" datasets.  This is likely to be the approach we take as we remove the code management for custom datasets.

So I think it's good to keep this condition in as a guard.